test/karma.conf.js

```diff
@@ -1,196 +1,41 @@
-module.exports = function(config) {
-  // build out a name for browserstack
-  // {TRAVIS_BUILD_NUMBER} [{TRAVIS_PULL_REQUEST} {PR_BRANCH}] {TRAVIS_BRANCH}
-  var browserstackName = process.env.TRAVIS_BUILD_NUMBER;
-
-  if (process.env.TRAVIS_PULL_REQUEST !== 'false') {
-    browserstackName += ' ' + process.env.TRAVIS_PULL_REQUEST + ' ' + process.env.TRAVIS_PULL_REQUEST_BRANCH;
-  }
-
-  browserstackName +=  ' ' + process.env.TRAVIS_BRANCH;
-
-  // Creating settings object first so we can modify based on travis
-  var settings = {
-    basePath: '',
-
-    frameworks: ['browserify', 'qunit', 'detectBrowsers'],
-    autoWatch: false,
-    singleRun: true,
-
-    // Compling tests here
-    files: [
-      '../dist/video-js.css',
-      '../test/globals-shim.js',
-      '../test/unit/**/*.js',
-      '../build/temp/browserify.js',
-      '../build/temp/webpack.js',
-      { pattern: '../src/**/*.js', watched: true, included: false, served: false }
-    ],
-
-    preprocessors: {
-      '../test/**/*.js': [ 'browserify' ]
-    },
-
-    browserify: {
-      debug: true,
-      plugin: ['proxyquireify/plugin'],
-      transform: [['babelify', {
-        "presets": [
-          [
-            "@babel/preset-env",
-            {
-              "loose": true
-            }
-          ]
-        ]
-      }]]
-    },
-
-    plugins: [
-      'karma-qunit',
-      'karma-chrome-launcher',
-      'karma-firefox-launcher',
-      'karma-ie-launcher',
-      'karma-opera-launcher',
-      'karma-safari-launcher',
-      'karma-safaritechpreview-launcher',
-      'karma-browserstack-launcher',
-      'karma-browserify',
-      'karma-coverage',
-      'karma-detect-browsers',
-    ],
-
-    detectBrowsers: {
-      enabled: false,
-      usePhantomJS: false
-    },
-
-    reporters: ['dots'],
-
-    // web server port
-    port: 9876,
-
-    // cli runner port
-    runnerPort: 9100,
-    colors: true,
-    logLevel: config.LOG_INFO,
-    captureTimeout: 300000,
-    browserNoActivityTimeout: 300000,
-    browserDisconnectTimeout: 300000,
-    browserDisconnectTolerance: 3,
-
-    browserStack: {
-      project: 'Video.js',
-      name: browserstackName,
-      build: browserstackName,
-      pollingTimeout: 30000,
-      captureTimeout: 600,
-      timeout: 600
-    },
-    customLaunchers: getCustomLaunchers(),
+const generate = require('videojs-generate-karma-config');
 
-    // The HTML reporter seems to be busted right now, so we're just using text in the meantime
-    // along with the summary after the test run.
-    coverageReporter: {
```

Comment:
we should figure out how to enable coverage again. Previously we've used coveralls but I have no qualms in switching to codecov